app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt

```diff
@@ -261,46 +282,155 @@ object GoogleMapHelper {
+                null,
+                true,
+                R.drawable.ic_downarrow,
+                0,
+                0,
+                100,
+                100,
+                100,
+                100
+            )
         )
         moveDownPos = moveDownMarker!!.position
+
         moveMarker = map!!.addMarker(
-            MarkerOptions().position(posCenter).icon(getMarkerRessource(R.drawable.ic_move))
-                .anchor(0.5f, 0.5f).draggable(true).snippet(
-                PolygonAction.MOVE.toString()
+            newMarker(
+                posCenter,
+                0.5f,
+                0.5f,
+                PolygonAction.MOVE.toString(),
+                null,
+                true,
+                R.drawable.ic_move,
+                0,
+                0,
+                100,
+                100,
+                100,
+                100
             )
         )
         movePos = moveMarker!!.position
+
+        rotationMarker = map!!.addMarker(
+            newMarker(
+                pos4,
+                0.5f,
+                0.5f,
+                PolygonAction.ROTATE.toString(),
+                null,
+                true,
+                R.drawable.ic_rotation,
+                0,
+                0,
+                100,
+                100,
+                100,
+                100
+            )
+        )
+        rotationPos = rotationMarker!!.position
+    }
+
+    /**
+     * Sets the values for the markers
+     */
+    fun newMarker(
+        pos: LatLng,
+        hAnchor: Float,
+        vAnchor: Float,
+        snippet: String?,
+        title: String?,
+        draggable: Boolean,
+        idDrawable: Int,
+        leftBound: Int,
+        topBound: Int,
+        rightBound: Int,
+        bottomBound: Int,
+        width: Int,
+        height: Int
+    ): MarkerOptions {
+        var mo = MarkerOptions().position(pos).anchor(hAnchor, vAnchor).draggable(draggable)
+            .snippet(snippet).title(title)
+        if (context != null) {
+            mo = mo.icon(
+                getMarkerRessource(
+                    idDrawable,
+                    leftBound,
+                    topBound,
+                    rightBound,
+                    bottomBound,
+                    width,
+                    height
+                )
+            )
+        }
+
+        return mo
     }
 
     /**
      * Generates the icon for the invisible icons
      * ref : https://stackoverflow.com/questions/35718103/how-to-specify-the-size-of-the-icon-on-the-marker-in-google-maps-v2-android
      * TODO : Check if we should make it a singleton to save memory/performance
      */
-    private fun getMarkerRessource(id: Int): BitmapDescriptor {
+    private fun getMarkerRessource(
+        id: Int,
+        leftBound: Int,
+        topBound: Int,
+        rightBound: Int,
+        bottomBound: Int,
+        width: Int,
+        height: Int
+    ): BitmapDescriptor {
```

Comment:
Same comment as above, you could maybe try to make an object "BoundAndDimension" or something of this kind so that we only get one parameter instead of 6

Reply:
Fixed by Mathieu's PR